sdk/src/main/scala/com/horizen/AbstractSidechainNodeViewHolder.scala

```diff
@@ -245,22 +247,100 @@ abstract class AbstractSidechainNodeViewHolder[
                                     stateToApply: MS,
                                     walletToApply: VL,
                                     suffixTrimmed: IndexedSeq[PMOD],
-                                    progressInfo: ProgressInfo[PMOD]): Try[SidechainNodeUpdateInformation]
+                                    progressInfo: ProgressInfo[PMOD]): Try[SidechainNodeUpdateInformation] =  Try {
+
+    val updateInfoSample = SidechainNodeUpdateInformation(history, stateToApply, walletToApply, None, None, suffixTrimmed)
+
+    val wrappedResultObj : Try[SidechainNodeUpdateInformation] = progressInfo.toApply.foldLeft[Try[SidechainNodeUpdateInformation]](Success(updateInfoSample)) {
```

Comment:
instead of `wrappedResultObj`,
```
 case (f@Failure(ex), _) =>
        log.error("Reporting modifier failed", ex)
        f
      case (success@Success(updateInfo), modToApply) =>
```
and `wrappedResultObj.get`

you may use `recoverWith`

Reply:
Ok,I used it for recovering any unexpected exception